x/marker/keeper/keeper.go

```diff
@@ -318,6 +318,29 @@ func (k Keeper) SetNetAssetValue(ctx sdk.Context, marker types.MarkerAccountI, n
 	return nil
 }
 
+// SetNetAssetValueWithBlockHeight adds/updates a net asset value to marker with a specific block height
+func (k Keeper) SetNetAssetValueWithBlockHeight(ctx sdk.Context, marker types.MarkerAccountI, netAssetValue types.NetAssetValue, source string, blockHeight uint64) error {
+	netAssetValue.UpdatedBlockHeight = blockHeight
+	if err := netAssetValue.Validate(); err != nil {
+		return err
+	}
+
+	setNetAssetValueEvent := types.NewEventSetNetAssetValue(marker.GetDenom(), netAssetValue.Price, netAssetValue.Volume, source)
```

Comment:
@iramiller Any thoughts on if we should still emit an event here?  We can either remove it, leave it as is, or add a new optional property to net asset value typed event of the height.

Reply:
I suppose we can leave it in there ... the block of events for changes in the upgrade is still useful even if the data set applies to a range of dates/blockheights.